src/trodes_to_nwb/convert_optogenetics.py

```diff
@@ -32,16 +32,29 @@ def add_optogenetics(nwbfile: NWBFile, metadata: dict, device_metadata: List[dic
         List of dictionaries containing metadata for devices used in the experiment.
     """
     logger = logging.getLogger("convert")
-    if not all(
-        [
-            x in metadata
-            for x in [
-                "virus_injection",
-                "opto_excitation_source",
-                "optical_fiber",
-                "optogenetic_stimulation_software",
+    if not (
+        all(
+            [
+                x in metadata
+                for x in [
```

Comment:
Might be a little cleaner to make this a tuple rather than a list and declare it outside the loop. Especially since it is used twice.